core/src/main/java/nl/nn/adapterframework/filesystem/MailFileSystemUtils.java

```diff
@@ -88,4 +101,40 @@
 		}
 	}
 	
+
+	public static String findBestReplyAddress(String from, String replyTo, String sender, Map<String,Object> headers) {
+		String result;
+		
+		if (null != (result = getValidAddress("replyTo", replyTo))) {
+			return result; 
+		}
+		if (null != (result = getValidAddress("from", from))) {
+			return result; 
+		}
+		if (null != (result = getValidAddress("sender", sender))) {
+			return result; 
+		}
+		Object returnPath = headers.get(RETURN_PATH_HEADER);
+		if (returnPath instanceof String && null != (result = getValidAddress("return path", (String)returnPath))) {
+			return result; 
+		}
+		return null;
+	}
+	
+	public static String getValidAddress(String type, String address) {
+		try {
+			if (StringUtils.isEmpty(address)) {
+				return null;
+			}
+			InternetAddress[] addresses = InternetAddress.parseHeader(address, true);
+			if (addresses.length==0) {
+				return null;
+			} 
+			return InternetAddress.toString(addresses);
+		} catch (AddressException e) {
+			log.warn("type ["+type+"] address ["+address+"] is invalid: "+e.getMessage());
```

Comment:
type en address omdraaien? Het is een adres-type niet een type-adres

_address [xyz] of type [abc] is invalid_, vs _type [abc] address [xyz] is invalid_